### DISCUSSION TITLE
How to troubleshoot locally

### DISCUSSION THREAD

**Discussion by smutel** [goreleaser/goreleaser#2813] [Q&A] [answered]
Hello,

I setup a github actions on my project https://github.com/smutel/terraform-provider-netbox/blob/6d669b494d956649cbfc69449c3774bda0aeddbb/.github/workflows/release.yml#L42.

I am using a custom CHANGELOG generated on the fly.
Previously these code worked correctly.

But now the release is created in github, the artifacts are created but the changelog is empty.

I would like to troubleshoot this issue but without doing it on github but on my laptop.

Is-it possible ?
Or another solution to help me to troubleshoot ?

**Comment by caarlos0** [comment, marked as answer]
Likely your `/tmp/RELEASE_CHANGELOG.md` is empty.

I noticed you're missing the `depth` argument in the checkout action... which is likely the cause (doing a shallow clone) 

That said, its not so easy to completely emulate github actions locally... and each case is different, so I can't help you with that

**Reply by smutel** [reply]
I did this after checkout:
```yaml
      - name: Unshallow
        run: git fetch --prune --unshallow
```

Not enough ?

**Reply by caarlos0** [reply]
not sure, I always use fetch-depth option, as its recommended here: https://goreleaser.com/ci/actions/

in any case, if the changelog is empty, your tool is likely generating an empty file... you could cat it after generating to make sure

**Reply by caarlos0** [reply]
https://github.com/goreleaser/goreleaser/pull/2885 this should help a bit
